src/pip/_internal/models/scheme.py

```diff
@@ -5,27 +5,21 @@
 https://docs.python.org/3/install/index.html#alternate-installation.
 """
 
+from dataclasses import dataclass
 
 SCHEME_KEYS = ["platlib", "purelib", "headers", "scripts", "data"]
 
 
+@dataclass
```

Comment:
Why can we not freeze this class?

Reply:
https://github.com/pypa/pip/blob/31437d675abfa1563cffb3f68a0a23f3cbd68afa/src/pip/_internal/locations/_sysconfig.py#L194-L197

Although this is trivial to fix... so I did that in https://github.com/pypa/pip/pull/12571/commits/3738d254100790348c0f6fc538dc8e1743df297e. Hopefully it doesn't break the test suite.